kk/utils/drf_enum_field.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from django.utils import six
+from django.utils.encoding import force_text
+from rest_framework.fields import ChoiceField
+
+
+class EnumField(ChoiceField):
+
+    def __init__(self, **kwargs):
+        self.enum_type = kwargs.pop("enum_type")
+        kwargs["choices"] = self.enum_type.choices()
+        super(EnumField, self).__init__(**kwargs)
+
+    def to_internal_value(self, data):
+        # Normal logic:
+        for choice in self.enum_type:
+            if choice.name == data or choice.value == data:
+                return choice
+
+        # Case-insensitive logic:
+        l_data = force_text(data).lower()
+        for choice in self.enum_type:
+            if choice.name.lower() == l_data or force_text(choice.value).lower() == data:
```

Comment:
Oy, good catch. (As it happens this method is more or less the only one that's not test-covered yet...)